### PR TITLE
Add length check when insert, upsert and check default value max length

### PR DESCRIPTION
**Pull request by smellthemoon**
Signed-off-by: lixinguo [xinguo.li@zilliz.com](mailto:xinguo.li@zilliz.com)
/kind bug
#24634

### CODE REVIEW

**Comment by smellthemoon** [comment]
rerun ut

File: internal/proxy/validate_util.go [hunk 1]
```diff
@@ -265,7 +265,15 @@ func (v *validateUtil) checkVarCharFieldData(field *schemapb.FieldData, fieldSch
...
 		}
-		return verifyLengthPerRow(strArr, maxLength)
+		if fieldSchema.GetDefaultValue() != nil {
+			defaultValueLength := len(fieldSchema.GetDefaultValue().GetStringData())
+			if int64(defaultValueLength) > maxLength {
+				msg := fmt.Sprintf("the length (%d) of string exceeds max length (%d)", defaultValueLength, maxLength)
+				return merr.WrapErrParameterInvalid("valid length string", "string length exceeds max length", msg)
+			}
+		} else {
+			return verifyLengthPerRow(strArr, maxLength)
```
Comment (yah01) [inline internal/proxy/validate_util.go:275]: Will this skip checking the length of data if the field with default?

File: pkg/util/parameterutil.go/get_max_len.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package proxy
+package parameterutil
```
Comment (yah01) [inline pkg/util/parameterutil.go/get_max_len.go:1]: utils should be with a short name, or no one is willing to use it, -> `paramutil`, but this should be in `typeutil` package

**Review by yah01 — APPROVED** [review]
/lgtm

**Pull request merged**